flask/app/variants.py

```diff
@@ -227,17 +226,109 @@ def parse_gene_panel() -> List[Any]:
+    matches = [region_pattern.match(r) for r in region.split(",")]
+
+    app.logger.info("Requested region: %s", region)
+
+    if not all(match is not None and len(match.groups()) == 3 for match in matches):
+        app.logger.error("Invalid region format: %s", region)
+        abort(400, description="Invalid region format")
+
+    # prepare set of regions
+    region_set = set([match.groups() for match in matches])
+    for r in region_set:
+        # fix regions to be consistent (start-end)
+        if int(r[START]) > int(r[END]):
+            r = (r[CHR], r[END], r[START])
+
+    # query
+    region_filter = or_(
+        *[
+            and_(
+                models.Variant.chromosome == tup[CHR],
+                int(tup[START]) <= models.Variant.position,
+                models.Variant.position <= int(tup[END]),
+            )
+            for tup in region_set
+        ]
+    )
+
+    query = models.Variant.query.filter(region_filter)
+
+    found_variants = query.with_entities(
+        func.count(distinct(models.Variant.variant_id))
+    ).scalar()
+    if found_variants == 0:
+        app.logger.warning("No requested variants were found")
+
+    return region_filter
+
+
+def parse_position(position: str):
+    """
+    Parses a given list of positions and chromosomes (eg. "chr1:5000,chrY:8000").
+    Returns a filter for use against the variant table to find corresponding variants
+    based on gene start and end positions.
+    """
+    CHR, POS = 0, 1
+    # format check
+    # 1: chromosome, 2: position
+    position_pattern = re.compile("chr([\da-zA-Z]+):([\d]+)")
+    matches = [position_pattern.match(p) for p in position.split(",")]
+
+    app.logger.info("Requested position: %s", position)
+
+    if not all(match is not None and len(match.groups()) == 2 for match in matches):
+        app.logger.error("Invalid position format: %s", position)
+        abort(400, description="Invalid position format")
+
+    # prepare set of positions
+    position_set = set([match.groups() for match in matches])
+
+    # query
+    position_filter = or_(
+        *[
+            and_(
+                models.Variant.chromosome == tup[CHR].upper(),
+                models.Variant.position == tup[POS],
+            )
+            for tup in position_set
+        ]
+    )
+
+    query = models.Variant.query.filter(position_filter)
+    found_variants = query.with_entities(
+        func.count(distinct(models.Variant.variant_id))
+    ).scalar()
+    if found_variants == 0:
+        app.logger.warning("No requested variants were found")
 
-    return ensgs
+    return position_filter
 
 
 @variants_blueprint.route("/api/summary/<string:type>", methods=["GET"])
 @login_required
 def summary(type: str):
     """
-    GET /api/summary/participants\?panel=ENSG00000138131
-    GET /api/summary/variants\?panel=ENSG00000138131
+    GET /api/summary/participants\?panel=ENSG00000138131&search_by=gene
```

Comment:
I was originally going to go with this, but decided on the current design to indicate that you can only search by one category at a time. I'll go ahead with this change though because I agree that it looks more intuitive